formfactory/tests/test_views.py

```diff
@@ -251,3 +251,18 @@ def test_form_actions_at_done_step(self):
 
     def tearDown(self):
         pass
+
+
+class FormHasRequestObjectTestCase(TestCase):
+    """Formfactory views should pass request to forms
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        load_fixtures(cls)
+
+    def test_request_is_passed_to_forms(self):
+        response = self.client.get(
+            reverse("formfactory:form-detail", args=[self.simpleform.slug]),
+        )
+        self.assertTrue(hasattr(response.context["form"], "request"))
```

Comment:
Need another test here please to ensure the request is instance of `rest_framework.request.Request` since you have `self.request = kwargs.pop("request", None)`, this would pass even if the request is None.